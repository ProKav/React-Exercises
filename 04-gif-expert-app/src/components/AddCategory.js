import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('')

	const handleInputChange = e => {
		setInputValue(e.target.value)
	}

	const handleSubmit = e => {
		e.preventDefault()

		if (inputValue.trim().length > 2) {
			setCategories(categ => [inputValue, ...categ])
			setInputValue('')
		} else {
			setInputValue('')
		}
	}

	return (
		<form className='form__group field' onSubmit={handleSubmit}>
			<br />
			<p>{inputValue}</p>
			<input
				className='form__field'
				name='name'
				id='name'
				type='text'
				value={inputValue}
				onChange={handleInputChange}
			/>
			<label htmlFor='name' className='form__label'>
				Type Here...
			</label>
		</form>
	)
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
}
