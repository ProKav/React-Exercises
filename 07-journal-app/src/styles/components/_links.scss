.link {
	color: $dark-grey;
	text-decoration: none;
	padding: 10px;

	&:hover {
		text-decoration: underline;
	}
}
