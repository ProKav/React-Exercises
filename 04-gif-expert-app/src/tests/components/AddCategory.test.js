import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('pruebas en <AddCategory />', () => {
	const setCategories = jest.fn()
	let wrapper = shallow(<AddCategory setCategories={setCategories} />)

	beforeEach(() => {
		jest.clearAllMocks()
		wrapper = shallow(<AddCategory setCategories={setCategories} />)
	})

	test('Debe mostrar <AddCategory /> correctamente', () => {
		expect(wrapper).toMatchSnapshot()
	})

	test('Debe de cambiar la caja de texto', () => {
		const input = wrapper.find('input')
		const value = 'Hola mundo'

		input.simulate('change', { target: { value } })
		expect(wrapper.find('p').text().trim()).toBe(value)
	})

	test('No debe postear la informacion On Submit', () => {
		wrapper.find('form').simulate('submit', { preventDefault() {} })
		expect(setCategories).not.toHaveBeenCalled()
	})

	test('Debe limpiar el testCategories y limpiar la caja de texto', () => {
		const value = 'Hello World'

		wrapper.find('input').simulate('change', { target: { value } })
		wrapper.find('form').simulate('submit', { preventDefault() {} })

		// expect(setCategories).toHaveBeenCalled()
		// expect(setCategories).toHaveBeenCalledWith( expect.any(Function)) // Evalua que sea una funcion lo que se llama
		expect(setCategories).toHaveBeenCalledTimes(1)
		expect(wrapper.find('input').prop('value')).toBe('')
	})
})
