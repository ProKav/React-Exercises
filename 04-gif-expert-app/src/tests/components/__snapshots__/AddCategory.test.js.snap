// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AddCategory /> Debe mostrar <AddCategory /> correctamente 1`] = `
<form
  className="form__group field"
  onSubmit={[Function]}
>
  <br />
  <p />
  <input
    className="form__field"
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    className="form__label"
    htmlFor="name"
  >
    Type Here...
  </label>
</form>
`;
